python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -1577,22 +1590,42 @@ def plot_events(self):
             return
 
         # Read the data.
-        result = self.reader.read(message_types=[EventNotificationMessage], remove_nan_times=False, **self.params)
-        data = result[EventNotificationMessage.MESSAGE_TYPE]
+        result = self.reader.read(message_types={MessageType.EVENT_NOTIFICATION} | COMMAND_MESSAGES | RESPONSE_MESSAGES,
+                                  remove_nan_times=False, return_in_order=True, return_bytes=True, **self.params)
+        data = result[None]  # In-order reads return data with the key None.
 
         if len(data.messages) == 0:
             self.logger.info('No event notification data available.')
             return
 
         table_columns = ['Relative Time (s)', 'System Time (s)', 'Event', 'Flags', 'Description']
-        table_data = [[], [], [], [], []]
-        table_data[0] = [f'{(m.system_time_ns - self.reader.get_system_t0_ns()) / 1e9:.3f}' for m in data.messages]
-        table_data[1] = [f'{(m.system_time_ns) / 1e9:.3f}' for m in data.messages]
-        table_data[2] = [str(m.action) for m in data.messages]
-        table_data[3] = [f'0x{m.event_flags:016X}' for m in data.messages]
-        table_data[4] = [m.event_description.decode('utf-8') for m in data.messages]
-
-        table_html = _data_to_table(table_columns, table_data)
+        rows = []
+        system_t0_ns = self.reader.get_system_t0_ns()
+        for message, message_bytes in zip(data.messages, data.messages_bytes):
+            system_time_ns = message.get_system_time_ns()
+            if isinstance(message, EventNotificationMessage):
+                event_type = message.event_type
+                flags = message.event_flags
+                description_str = EventNotificationMessage.event_description_to_string(message.event_type,
```

Comment:
Based on how this is being used, it seems like this would make more sense as a member function? I guess I find the usage here a bit more confusing then:

```python
if isinstance(message, EventNotificationMessage):
                event_type = message.event_type
                flags = message.event_flags
                description_str = message.event_description_to_string()
            else:
                flags = None
                description_str = repr(message)
```

This seems simpler then your current implementation that re-unpacks the known command types.

Reply:
Yeah might need to think about this a bit. I agree a member function would be simpler. For the `else` case though, I wanted it to use the same format as a command packed within an event (just with some things missing if not applicable like system time). The current version looks like:
```
[GET_CONFIG (13101), source=ACTIVE, type=ENABLED_GNSS_FREQUENCY_BANDS]
Data (28 B): 2E 31 00 00 E9 06 06 E0 02 00 2D 33 00 00 00 00 04 00 00 00 00 00 00 00 33 00 00 00
```

The `repr()` doesn't have the data bytes portion. I suppose I could just implement it here though.

Reply:
Why though? Some of the messages are KB large? It seems like you'd only want that for corrupted messages which would never make it here.

Reply:
Corrupted messages will still be printed here, just not the decoded contents. The main use for this is diagnosing issues if a user gets an unexpected response or sends an incorrectly formed command.

At KB sized messages, this will definitely be quite big. I may truncate the printout in that case. That would mainly be for platform storage/export queries though. Right now, we explicitly do not output response events for those to avoid wasting bandwidth since platform storage is automatically output for any interfaces with diagnostic data enabled.